service-oriented-architecture/src/main/java/com/iluwatar/soa/security/WebSecurityConfig.java

```diff
@@ -0,0 +1,60 @@
+package com.iluwatar.soa.security;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configurers.LogoutConfigurer;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.provisioning.InMemoryUserDetailsManager;
+import org.springframework.security.web.SecurityFilterChain;
+
+
+@Configuration
+@EnableWebSecurity
+public class WebSecurityConfig {
+
+  @Bean
+  public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+    http
+        .authorizeHttpRequests(requests -> requests
+            .requestMatchers("/login").permitAll()
+            .anyRequest().authenticated()
+        )
+        .formLogin(form -> form
+            .defaultSuccessUrl("/home/greeting", true)
+            .permitAll()
+        )
+        .logout(LogoutConfigurer::permitAll);
```

Comment:
LogoutConfigurer::permitAll  -- > ensure that only the logged-in user can initiate their own logout